docs/11_templating.md

```diff
@@ -0,0 +1,704 @@
+| `*.hbs` extension | Handlebars processing, `.hbs` extension stripped |
+| Path contains `{{var}}` | Variable substitution in path |
+| **All other files** | **Copied verbatim (default)** |
+
+You only need to specify transforms for files that require processing. Binary files, assets, and files with `{{}}` syntax that isn't meant to be templated will be safely copied without modification.
+
+---
+
+## Template Metadata Schema
+
+The `template.json` file defines everything needed for discovery, selection, and generation:
+
+```json
+{
+  "$schema": "../../schemas/template-v1.json",
+  "version": "1.0.0",
+  "type": "application",
+  "id": "my-template",
+  "displayName": "My Template",
+  "description": "What this template provides",
+  "extends": "ios-base",
+  
+  "platform": {
+    "type": "ios",
+    "minVersion": "15.0"
+  },
+  
+  "useCase": {
+    "primary": "Brief description of primary use case",
+    "scenarios": ["Scenario 1", "Scenario 2"],
+    "when": "When to choose this template"
+  },
+  
+  "capabilities": ["offline-sync", "contact-management"],
+  "tags": ["ios", "swift", "mobilesdk"],
+  
+  "templateVariables": [
+    {
+      "name": "projectName",
+      "type": "string",
+      "description": "The project name",
+      "required": true,
+      "validation": "^[A-Za-z][A-Za-z0-9_]*$"
+    }
+  ],
+  
+  "extensionPoints": [
+    {
+      "id": "add-feature",
+      "name": "Add Custom Feature",
+      "description": "How to add a custom feature",
+      "affectedFiles": ["path/to/file.swift"],
+      "aiGuidance": "Create a new model file following the existing pattern. Implement the required protocol. Register the feature in the main configuration."
+    }
+  ],
+  
+  "generation": {
+    "fileTransforms": [
+      { "pattern": "**/*.{hbs,swift}", "processor": "handlebars" },
+      { "pattern": "**/Podfile", "processor": "handlebars", "outputExtension": "" }
+      // All other files (images, assets, etc.) are copied as-is by default
+    ],
+    "fileOperations": [
+      { "action": "rename", "from": "ExampleApp", "to": "{{projectName}}" },
+      { "action": "delete", "from": ".DS_Store" }
+    ]
+  }
+}
+```
+
+### Key Metadata Fields
+
+| Field | Purpose |
+|-------|---------|
+| `extends` | Template ID to inherit from (optional) |
+| `platform` | Target platform and version constraints |
+| `capabilities` | Semantic tags for capability-based search |
+| `templateVariables` | User-provided values with validation |
+| `extensionPoints` | Guidance for post-generation customization |
+| `generation.fileTransforms` | File processing rules (opt-in for processing) |
+| `hidden` | If true, hide from discovery (for base templates) |
+
+### File Processor Types
+
+| Processor | Behavior | When to Use |
+|-----------|----------|-------------|
+| `handlebars` | Processes file content with Handlebars, substitutes `{{variables}}` | Text files that need variable substitution (source code, config files) |
```

Comment:
right now `handlebars` is the only processing option but I wanted to keep this flexible in case we ever wanted to introduce other strategies